src/components/AddCompanies/AddCompanies.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Form, FormGroup, ControlLabel, FormControl, Col } from 'react-bootstrap';
+import './styles.less';
+
+const propTypes = {
+    addCompany: PropTypes.func.isRequired,
+};
+
+class AddCompanies extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			name: '',
+			address: '',
+			phone: '',
+			revenue: '',
+		};
+
+		this.addNewCompany = this.addNewCompany.bind(this);
+		this.handleFormControlChange = this.handleFormControlChange.bind(this);
+	}
+
+	addNewCompany(e) {
+		e.preventDefault();
+		const newCompany = {...this.state};
+		this.props.addCompany(newCompany)
+	}
+
+	handleFormControlChange(e) {
+		const { value, id } = e.target;
+		console.log('VALUE:', value);
+		console.log('FOR ITEM WITH NAME:', id);
+		this.setState({
+			[id]: value,
+		});
+	}
+
+  render() {
+    return (
+      <div className="addCompanies_wrapper">
+        <Form horizontal id="test">
+					<FormGroup controlId="name">
+					  <ControlLabel>Name:</ControlLabel>
+					  <FormControl
+					    type="text"
+							value={this.state.name}
+					    placeholder="Company Name"
+					    onChange={this.handleFormControlChange}
+					  />
+					</FormGroup>
```

Comment:
I modified this section because again, I don't know react-bootstrap, so this was my best guess at how to do this best.
If you notice, these `FormGroup` component's all have the same pattern. So with a map function and a config I could save some lines of code by doing something like:
```javascript
/* this would go right below your imports in the file on like, line 5 */
const controlConfig = [
  {
    title: 'name',
    label: 'Name',
  },
  {
    title: 'address',
    label: 'Address',
  },
  {
    title: 'phone',
    label: 'Phone',
  },
  {
    title: 'revenue',
    label: 'Revenue',
  }
]
```
then in the class, you could have a method like:
```javascript
renderFormGroup() {
  return controlConfig.map(element => (
    <FormGroup controlId="address">
      <ControlLabel>{`${element.label}:`}</ControlLabel>
      <FormControl
        type="text"
        value={this.state[element.title]}
        placeholder=`Company ${element.label}`
        onChange={this.handleFormControlChange}
     />
  </FormGroup>
  ));
}```